buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/RandomizedTestingTask.groovy

```diff
@@ -95,6 +97,16 @@ class RandomizedTestingTask extends DefaultTask {
         listenersConfig.listeners.add(new TestReportLogger(logger: logger, config: testLoggingConfig))
     }
 
+    void basedOn(RandomizedTestingTask other) {
```

Comment:
I really would like to not have this. It makes many assumptions (eg that that `check` should depend on this). Can we please keep this change only to updating testClassesDirs and look at ways to deduplicate internClusterTest in a separate issue?

Reply:
Removed it. I have a better understanding of why this was not a good idea now, and why this would be better handled in it's own PR.  There are multiple ways to manage tests, deal with unit and integrations tests etc ( there's an `additionalTests` extension that is similar to what `basedOn` did ).  It might also be worth setting up a different source path for integration tests then we could auto wire that to check and offer a clear DSL for defining new tests that are possibly  a subset of the existing unit or integration with additional system properties.